src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2092,6 +2031,168 @@ WorkerExtendedOpNode(MultiExtendedOp *originalOpNode,
 }
 
 
+/*
+ * ProcessWorkerExpressionList is a utility function which processes the
+ * expressions that are intended to be added to the worker target list.
+ *
+ * In summary, the function gets a list of expressions, converts them to target
+ * entries and updates all the necessary fields such that the expression is correctly
+ * added to the worker query's target list.
+ *
+ * Adding an expression to the target list includes generating a target entry,
+ * appending it to the newTargetEntryList and incrementing the projection number.
+ *
+ * If the expression is requested to be added to the group clause, the entry
+ * is appended to groupClauseList and sort group reference is incremented.
+ */
+static void
+ProcessWorkerExpressionList(List *expressionList, TargetEntry *originalTargetEntry,
```

Comment:
I find the name a bit confusing. Would something like `ExpandWorkerTargetEntry` be appropriate?